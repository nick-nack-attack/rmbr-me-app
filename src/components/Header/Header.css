@media only screen and (min-width: 320px) {

    .Header h1 {
        margin: 0;
    }

    .Header a {
        color: white;
        text-decoration: none;
    }

    .Header a:hover {
        color: #58cfff;
    }

    .Header__tagline--wide {
        opacity: 0.35;
        margin-left: 10px;
    }

    .Header__tagline--narrow {
        opacity: 0.35;
        display: block;
        text-align: center;
        display: none;
        margin-top: 10px;
    }

    .Header__not-logged-in {
        display: flex;
        flex: 1;
        align-items: center;
        justify-content: flex-end;
    }

    .Header__not-logged-in a:last-child {
        background: transparent;
        color: white;
        font-weight: bold;
        padding: 5px 15px;
        border: 2px solid white;
        border-radius: 15px;
        margin-left: 10px;
    }

    .Header__not-logged-in a:last-child:hover {
        color: #CA1551;
        background-color: white;
    }

    .Header__logged-in a {
        border: 1px solid white;
        padding: 5px 15px;
        border-radius: 15px;
    }

    .Header h1 {
        font-size: 18pt;
    }

    .Header__tagline--wide {
        display: none;
    }

    .Header a {
        font-weight: bold;
    }

    .rmbr_add_details_button {
        text-align: left;
    }

}

@media only screen and (min-width: 550px) {

    .Header {
        align-items: baseline;
    }

    .Header__tagline--narrow {
        display: inherit;
    }

    .Header__tagline--wide {
        display: inline;
    }

    .Header h1 {
        font-size: 22pt;
    }

}

@media only screen and (min-width: 800px) {
    .Header {
        align-items: center;
    }
}