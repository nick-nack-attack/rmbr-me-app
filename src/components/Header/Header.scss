@import '../../themes/_v2.scss';

.Header {
    height: 70px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    background-color: $header;
    padding: 10px 30px;
    border-bottom: 2px solid $headerSecondary;

    &>:hover {
        cursor: pointer;
        filter: drop-shadow(0px 2px 2px black);
        // transition: width 0.4s ease-in-out;
        -webkit-transition: filter 0.5s ease-in-out;
        -moz-transition: filter 0.5s ease-in-out;
        -o-transition: filter 0.5s ease-in-out;
        transition: filter 0.5s ease-in-out;
    };

}

#rmbrme_logo {
    height: 40px;
    margin-top: 5px;
}
