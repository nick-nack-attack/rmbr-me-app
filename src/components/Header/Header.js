import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { Hyph } from "../Utils/Utils";
import TokenService from '../../services/token-service'
import IdleService from "../../services/idle-service";
import './Header.css'

class Header extends Component {

    handleLogoutClick = () => {
        TokenService.clearAuthToken()
        TokenService.clearCallbackBeforeExpiry()
        IdleService.unregisterIdleResets()
    };

    renderLogoutLink() {
        console.log('renderLogoutLink ran');
        return (
            <div className='Header__logged-in'>
                <Link
                    onClick={this.handleLogoutClick}
                    to='/'>
                    Logout
                </Link>
            </div>
        )
    };

    renderLoginLink() {
        console.log('renderLoginLink ran');
        return (
            <div className='Header__not-logged-in'>
                <Link
                    to='/register'>
                    Register
                </Link>
                <Hyph/>
                <Link
                    to='/login'>
                    Log in
                </Link>
            </div>
        )
    };

    render() {
        return <>
            <nav className='Header'>
                <h1>
                    <Link to={TokenService.hasAuthToken() ? '/my-people' : '/'}>
                        {' '}
                        rmbr me
                    </Link>
                </h1>
                <span className='Header__tagline--wide'>Be a Better Buddy</span>
                { TokenService.hasAuthToken()
                    ? this.renderLogoutLink()
                    : this.renderLoginLink() }
            </nav>

            <span className='Header__tagline--narrow'> </span>
        </>
    }
}

export default withRouter(Header)