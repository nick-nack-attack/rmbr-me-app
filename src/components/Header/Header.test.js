import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from "./Header";

describe(`Header component`, () => {

    it('renders Header by default', () => {
        const wrapper = shallow(<Header/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders the login / logout buttons', () => {
        const wrapper = shallow(<Header/>)
            .find('#log-in-link', '#sign-up-link')
                expect(toJson(wrapper)).toMatchSnapshot()
    })
})