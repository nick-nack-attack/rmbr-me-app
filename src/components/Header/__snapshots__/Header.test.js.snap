// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Header component renders the login / logout buttons 1`] = `null`;
